enterprise/kernel/src/test/java/org/neo4j/kernel/impl/enterprise/store/id/NodeIdReuseStressIT.java

```diff
@@ -120,7 +121,17 @@ private static void maybeRunIdMaintenance( GraphDatabaseService db, int iteratio
         if ( iteration % 100 == 0 && ThreadLocalRandom.current().nextBoolean() )
         {
             DependencyResolver resolver = dependencyResolver( db );
-            resolver.resolveDependency( IdController.class ).maintenance();
+            IdController idController = resolver.resolveDependency( IdController.class );
+            if ( idController != null )
+            {
+                idController.maintenance();
+            }
+            else
+            {
+                System.out.println( "Id controller is null. Dumping resolver content." );
+                System.out.println( "Resolver: " + ReflectionToStringBuilder.toString( resolver ) );
```

Comment:
I tried and you will fill the log with lines like this for every single page in the page cache:
```
Page[ id = 134, address = 0, filePageId = -1, swapperId = 0, usageCounter = 0 ] OffHeapPageLock[Flush: 0, Excl: 1, Mod: 0, Ws: 0, S: 0]
```
Maybe filter those out, or if you are fine with that then let's go ahead. I'll put "require change".

Reply:
yeah i know, TC should be able to survive that right?